File: problems/Swift/495.提莫攻击.swift
/**
 * 495.提莫攻击 [Easy]
 * Time Complexity: O(n), Space Complexity: O(1)
 * https://leetcode-cn.com/problems/teemo-attacking/
 */

class Solution {
  func findPoisonedDuration(_ timeSeries: [Int], _ duration: Int) -> Int {
    var res = 0, expired = 0

    for time in timeSeries {
      if time > expired {
        res += duration
      } else {
        res += time + duration - expired
      }

      expired = time + duration
    }

    return res
  }
}

File: problems/rust-solutions/src/495.提莫攻击.rs
// 495.提莫攻击 [Easy]
// Time Complexity: O(n), Space Complexity: O(n)
// hhttps://leetcode-cn.com/problems/teemo-attacking/
impl Solution {
  pub fn find_poisoned_duration(time_series: Vec<i32>, duration: i32) -> i32 {
    let mut res = 0;
    let mut expired = 0;

    for time in time_series {
      if time > expired {
        res += duration;
      } else {
        res += time + duration - expired
      }

      expired = time + duration;
    }

    return res;
  }
}
